src/test/regress/sql/with.sql

```diff
@@ -1065,3 +1065,40 @@ WITH cte1 AS (
 SELECT * FROM cte1 a JOIN cte1 b USING (c1);
 
 DROP TABLE t1;
+
+-- Test join prefetch_inner in the case of bottleneck join
+SET optimizer = off;
+--start_ignore
+DROP TABLE IF EXISTS d;
+--end_ignore
+CREATE TABLE d (c1 int, c2 int) DISTRIBUTED BY (c1);
+
+INSERT INTO d (VALUES ( 2, 0 ),( 2 , 0 ));
+
+-- Test with HashJoin
+EXPLAIN (ANALYZE, COSTS OFF) WITH cte AS (
+	SELECT count(*) c1 FROM d
+) SELECT * FROM cte a JOIN (SELECT * FROM d JOIN cte USING (c1) LIMIT 1) b USING (c1);
+
+-- Test that the logic for Nested loop and Merge Join has not changed.
+SET enable_hashjoin = off;
+SET enable_nestloop = on;
```

Comment:
What are we testing here?  That nothing hash changed? 
@RekGRpth  asked to write test cases for both chagnes, but for what?

If we want to really test the prefetch then we should leave outer part empty to see that inner part is (never executed) for Nest Loop and Merge Join. And is executed for HashJoin

Otherwise there is no sense to even add these tests

Reply:
Or we want to show that Shared Scans at various BottleNeck joins don't cause a deadlock?

Reply:
I think this makes sense since all joins types can get a deadlock in case of an incorrect prefetch_inner. I think it's worth creating test cases for them.

Reply:
We are solving the deadlock problem for Shared Scans in case of Hash Join. Other types are not affected as you said. 

And how we get the deadlock for Bottleneck joins without Shared Scans?